### PR TITLE
Update NetApp files module

### PR DESCRIPTION
**Pull request by Msanzdelrio**
# Change

Updated NetApp files module to the new child-resource structure.
## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.NetApp/netAppAccounts/.parameters/parameters.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
                             // NFS41 VOL
                             "poolVolumeName": "vol01-nfsv41",
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/.parameters/parameters.json:50]: Should this be `name `too?

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+module capacityPool_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${deployment().name}-Rbac-${index}'
+  params: {
+    roleAssignmentObj: roleAssignment
+    resourceName: capacityPool.name
+  }
+}]
+
+@description('The name of the Capacity Pool.')
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/deploy.bicep:86]: Please add new lines between outputs

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+module volume_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${deployment().name}-Rbac-${index}'
+  params: {
+    roleAssignmentObj: roleAssignment
+    resourceName: volume.name
+  }
+}]
+
+@description('The name of the Volume.')
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep:69]: Please add new lines between outputs

File: arm/Microsoft.NetApp/netAppAccounts/deploy.bicep [hunk 4]
```diff
@@ -85,15 +85,24 @@ module netAppAccount_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, ind
...
+    size: capacityPool.size
+    serviceLevel: contains(capacityPool, 'serviceLevel') ? capacityPool.serviceLevel : 'Standard'
+    qosType: contains(capacityPool, 'qosType') ? capacityPool.qosType : 'Auto'
+    volumes: contains(capacityPool, 'volumes') ? capacityPool.volumes : []
+    coolAccess: contains(capacityPool, 'coolAccess') ? capacityPool.coolAccess : false
+    roleAssignments: contains(capacityPool, 'roleAssignments') ? capacityPool.roleAssignments : []
   }
 }]
 
+@description('The name of the NetApp account.')
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/deploy.bicep:103]: Please add new lines between outputs

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep [hunk 5]
```diff
@@ -0,0 +1,74 @@
+@description('Required. The name of the capacity pool.')
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep:1]: Should be `The name of the NetApp account.`

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/deploy.bicep [hunk 6]
```diff
@@ -0,0 +1,91 @@
+@description('Required. The name of the capacity pool.')
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/deploy.bicep:1]: Should be `The name of the NetApp account.` Requires a readme update

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/deploy.bicep [hunk 7]
```diff
@@ -46,46 +46,52 @@ module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
...
-  location: location
-  tags: tags
-  properties: {
-    serviceLevel: serviceLevel
-    size: size
-    qosType: qosType
-    coolAccess: coolAccess
+resource netAppAccount 'Microsoft.NetApp/netAppAccounts@2021-04-01' existing = {
+  name: netAppAccountName
+  resource capacityPool 'capacityPools@2021-06-01' = {
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/deploy.bicep:51]: This nested approach is still under discussion (ref. pinned discussion in the internal channel). Not sure if worthy to refactor the module at this level already. @MrMCake thoughts? I sense we'll end up keeping the main resource at top level.

Comment (MrMCake) [reply arm/Microsoft.NetApp/netAppAccounts/capacityPools/deploy.bicep:51]: I'd personally prefer the nesting, but it's as we said in the channel - not agreed upon. For now we have different approaches merged. I.e. in storage account we have the nesting, in automation account the nesting aside from the top level resource, and most if not all others use neither but instead string-concat. 

Hence I'd not have a strong opinion about it and would be fine either way.

However - we we keep it as is @Msanzdelrio then please update line 87 & reference `netApAccount.name`

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+param cuaId string = ''
+
+module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource netAppAccount 'Microsoft.NetApp/netAppAccounts@2021-04-01' existing = {
+  name: netAppAccountName
+  resource capacityPool 'capacityPools@2021-06-01' existing = {
```
Comment (MrMCake) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep:50]: add an empty row above to show the separation

Comment (MrMCake) [reply arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep:50]: same below

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+}
+
+module volume_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${deployment().name}-Rbac-${index}'
+  params: {
+    principalIds: roleAssignment.principalIds
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    resourceName: '${netAppAccountName}/${netAppAccount::capacityPool.name}/${netAppAccount::capacityPool::volume.name}'
```
Comment (MrMCake) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/deploy.bicep:72]: Change `netAppAccountName` to `netAppAcount.name`

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/readme.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
+# AzureNetAppFilesCapacitypoolsVolumes `[Microsoft.NetApp/netAppAccounts/capacityPools/volumes]`
```
Comment (MrMCake) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/volumes/readme.md:1]: Update the name to `Azure NetApp Files Capacity Pool Volumes`

**Review by MariusStorhaug — APPROVED** [review]
No further comments from me, and other comments are resolved. Approving.

**Pull request merged**